### PR TITLE
no need to generate service when href defined

### PR DESCRIPTION
**Pull request by AngryBerryMS**
[AB#347152](https://dev.azure.com/ceapex/Engineering/_workitems/edit/347152?src=WorkItemMention&src-action=artifact_link)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6886)

### CODE REVIEW

File: src/docfx/build/toc/ServicePageGenerator.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void GenerateServicePageFromTopLevelTOC(TocNode node, List<FilePath> resu
             var referenceTOCFullPath = Path.GetFullPath(Path.Combine(_docsetPath, referenceTOCRelativeDir));
             var nodeHrefFullPath = Path.GetFullPath(Path.Combine(referenceTOCFullPath, node.Href.Value ?? ""));
 
-            if (string.IsNullOrEmpty(node.Href.Value) || !File.Exists(nodeHrefFullPath))
```
Comment (herohua) [inline src/docfx/build/toc/ServicePageGenerator.cs:51]: This looks hack to me. Is it possible to determine by checking whether there is a file of that name already existing in the repo?

**Review by herohua — CHANGES_REQUESTED** [review]
prevent auto-merge

**Comment by herohua** [comment]
Talked with @AngryBerryMS and we prefer to consolidate the behavior as `We don't generate service page as long as href is being specified in top-level-toc.yml regardless of whether the specified file exists, or the file path starts with ~ or not`.

+ @live1206 @mingwli to comment

**Comment by AngryBerryMS** [comment]
Will fix after xiangwei and peican check the TOPLevelTOC of current repos to confirm whether relative path for item that contain landingpageType property

File: src/docfx/build/toc/ServicePageGenerator.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/ServicePageGenerator.cs:51]: Fount the way to get fullpath when start with ~

**Review by yufeih — APPROVED** [review]

**Pull request merged**